src/sagemaker_xgboost_container/checkpointing.py

```diff
@@ -42,72 +42,73 @@ def train(train_args, checkpoint_dir):
 
     xgb_model, start_iteration = load_checkpoint(checkpoint_dir)
 
+    # xgboost's default value for num_boost_round is 10.
+    # If num_boost_round <= 0, xgb.train() doesn't actually train and
+    # immediately returns a Booster object.
+    train_args["num_boost_round"] = train_args.get("num_boost_round", 10) - start_iteration
+
     if xgb_model is not None:
-        logging.info("Checkpoint loaded from %s", xgb_model)
-        logging.info("Resuming from iteration %s", start_iteration)
+        logging.warning("Checkpoint loaded from %s", xgb_model)
+        logging.warning("Resuming from iteration %s", start_iteration)
 
     callbacks = train_args.get("callbacks", [])
-    callbacks.append(print_checkpointed_evaluation(start_iteration=start_iteration))
-    callbacks.append(save_checkpoint(checkpoint_dir, start_iteration=start_iteration))
+    callbacks.append(print_checkpointed_evaluation(start_iteration=start_iteration,
+                                                   end_iteration=train_args["num_boost_round"]))
+    callbacks.append(save_checkpoint(checkpoint_dir, start_iteration=start_iteration, iteration=start_iteration,
+                                     end_iteration=train_args["num_boost_round"]))
 
     train_args["verbose_eval"] = False  # suppress xgboost's print_evaluation()
     train_args["xgb_model"] = xgb_model
     train_args["callbacks"] = callbacks
-    # xgboost's default value for num_boost_round is 10.
-    # If num_boost_round <= 0, xgb.train() doesn't actually train and
-    # immediately returns a Booster object.
-    train_args["num_boost_round"] = train_args.get("num_boost_round", 10) - start_iteration
 
     booster = xgb.train(**train_args)
 
     return booster
 
 
-def print_checkpointed_evaluation(period=1, show_stdv=True, start_iteration=0):
-    """Create a callback that print evaluation result.
```

Comment:
the comments were not supposed to be removed, I'll make sure of all the removed comments in the following revision